### DISCUSSION TITLE
Role-Based Access Control (RBAC)

### DISCUSSION THREAD

**Discussion by br3ndonland** [vercel/community#800] [Help] [answered]
[Role-Based Access Control (RBAC)](https://www.okta.com/identity-101/what-is-role-based-access-control-rbac/) is a method for controlling who can access what. Common use cases for implementing RBAC could include controlling access to the admin dashboard of a site, or controlling access to a web page that displays sensitive data.

Netlify has [docs](https://docs.netlify.com/visitor-access/role-based-access-control/) and [blogs](https://www.netlify.com/blog/2019/01/31/restrict-access-to-your-sites-with-role-based-redirects/) on how to implement this.

[Vercel does provide RBAC for team members](https://vercel.com/docs/concepts/teams/roles-and-permissions). However, this is just for teams developing on Vercel. It doesn't appear to apply to users viewing deployments.

What support does Vercel provide for RBAC in deployed applications? Does it depend entirely on the framework for this, such as [NextAuth.js](https://next-auth.js.org/) for Next.js?

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out @br3ndonland! You would need to implement your own role-based redirects using your preferred auth solution, such as [NextAuth.js](https://next-auth.js.org/). Our [authentication](https://vercel.com/docs/concepts/solutions/authentication) doc explains a few options and has a list of [authentication providers](https://vercel.com/docs/concepts/solutions/authentication#providers) you could use.
